src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineFlexibleStaging.java

```diff
@@ -73,33 +73,30 @@ public void stageFlexible(StageFlexibleConfiguration config) throws AppEngineExc
     }
   }
 
-  private static String findRuntime(StageFlexibleConfiguration config) {
-    // verification for app.yaml that contains runtime:java
-    Path appYaml = config.getAppEngineDirectory().toPath().resolve(APP_YAML);
-    String runtime = null;
+  @VisibleForTesting
+  static String findRuntime(StageFlexibleConfiguration config) throws IOException {
     try {
-      if (Files.isRegularFile(appYaml)) {
-        runtime = new AppYaml(appYaml).getRuntime();
-      }
-    } catch (IOException e) {
-      log.warning("Unable to determine runtime: error parsing app.yaml : " + e.getMessage());
+      // verification for app.yaml that contains runtime:java
+      Path appYaml = config.getAppEngineDirectory().toPath().resolve(APP_YAML);
+      return new AppYaml(appYaml).getRuntime();
+    } catch (ScannerException | ParserException ex) {
+      throw new AppEngineException("Malformed 'app.yaml'.", ex);
     }
-    return runtime;
   }
 
   @VisibleForTesting
   static void copyDockerContext(StageFlexibleConfiguration config, CopyService copyService,
       String runtime) throws IOException {
     if (config.getDockerDirectory() != null && config.getDockerDirectory().exists()) {
-      if (runtime != null && runtime.equals("java")) {
-        log.warning("WARNING: 'runtime 'java' detected, any docker configuration in "
+      if ("java".equals(runtime)) {
```

Comment:
It is different from appengine-web.xml. It is always "java" for Java. You can check this in the official app.yaml reference doc.